### ISSUE TITLE
Accessibility Problem on Media, Location and Fonts combobox

### ISSUE THREAD

**Issue by legle** [fisharebest/webtrees#1369]
Hello,

I am a blind user of the webtrees and I am not able to work with the combo box to select the images, locations and fonts and all the similar operating combobox.

I have already tried to select using the following combinations of screen readers and browsers:

• JAWS and Internet Explorer;
• JAWS and Google Chrome;
• NVDA and Internet Explorer;
• NVDA and Google Chrome;
• Voice Over and Safari on IPhone;
• Talkback and Chrome on Android;

At the moment I can not work with these controls.

Is there any way you can help me with this?

**Comment by fisharebest** [comment]
Is this webtrees version 1.7.9 or webtrees version 2.0.0-dev?

Fixing accessibility issues is a high priority.

My employer has a well-equipped laboratory for accessibility testing, so I have access to JAWS, NVDA, etc.  But I do not always have enough time to use it.

**Comment by legle** [comment]
My version of webtrees is 2.0.0-dev and I'm upgrading from the git repository.

The webtrees do not have many accessibility issues, but this particular problem should be fixed as soon as possible.

Sorry for my english, I'm using google translator, if you have trouble understanding what I say, please let me know.

**Comment by fisharebest** [comment]
We use a libarary called "select2" for these combo boxes.

The issue tracker for select2 lists many accessibility issues that have been open for a long time.

I will investigate other libraries.  One possibility is called "chosen".  It has a demo page (https://harvesthq.github.io/chosen/).

Can you visit this page and tell me if it works OK with JAWS/NVDA/etc.

**Comment by fisharebest** [comment]
The twitter/typeahead/bloudhound looks interesting, but I think the project was abandoned.

I have some colleagues that specialize in accessibility.  I will ask them.

**Comment by legle** [comment]
JAWS and NVDA can read these components, I can use the filter and I can select the combo box items.

In some cases I do not know if it was working correctly, but overall this plug-in seems to work.

works better than what works on webtrees today, but I noticed some problems:

In multiple choice fields I can not see what is selected;
In the fields where the item changes sideways I do not know if the item was added because this is not shown.

There are not many problems with this component. His developer could use a bit more aria-selected and other aria attributes, but for the time being it is better.

**Comment by legle** [comment]
Hello,

The componente used on webtrees 1.7.9 whorks with JAWS and NVDA.

The Webtrees 2.0 only run on php 7?

**Comment by fisharebest** [comment]
I can use JAWS and NVDA at work.  But work is very busy, and I have not had time.  I expect to have time for this towards the end of December.

webtrees 2.0 requires PHP 7.0 or higher.

**Comment by fisharebest** [comment]
There are three libraries available for this function: chosen, select2 and selectize.

All three project are semi-abandoned.  There is a discussion [here](https://github.com/select2/select2/issues/4953).

Chosen does not support remote data sources or mobile.

Selectize claims to work with a keyboard.  Need to investigate it.

**Comment by fisharebest** [comment]
Selectize has an open issue regarding lack of support for NVDA/JAWS https://github.com/selectize/selectize.js/issues/798

**Comment by fisharebest** [comment]
Sorry for not responding to this issue for a long time.

We changed to a new library many years ago.  I forgot to update this issue.
